nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -424,6 +423,40 @@ public static void checkTypeParameterNullnessForAssignability(
     if (tree instanceof VariableTree) {
       VariableTree varTree = (VariableTree) tree;
       rhsTree = varTree.getInitializer();
+      Tree lhsTypeTree = varTree.getType();
+
+      if (rhsTree instanceof MethodInvocationTree) {
+        MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+        Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+        // update inferredTypes cache for assignments
+        if (methodSymbol.type instanceof Type.ForAll // generic method call
+            && methodInvocationTree.getTypeArguments().isEmpty() // no explicit generic arguments
+            && lhsTypeTree instanceof ParameterizedTypeTree) { // lhs type has generic
+          if (lhsType != null) {
+            List<Symbol.TypeVariableSymbol> typeParam = methodSymbol.getTypeParameters();
+            List<Type> returnTypeTypeArg = methodSymbol.getReturnType().getTypeArguments();
+
+            // if generic type in return type
+            Map<Type, Type> genericNullness = new HashMap<>();
+            for (int i = 0; i < typeParam.size(); i++) {
+              Type upperBound = typeParam.get(i).type.getUpperBound();
+              // generic has nullable upperbound
+              if (getTypeNullness(upperBound, analysis.getConfig()) == Nullness.NULLABLE) {
+                Type lhsInferredType =
+                    inferMethodTypeArgument(
+                        typeParam.get(i).type,
```

Comment:
Store this expression in a local variable with a readable name, like `curTypeParam`, and re-use it